File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/SavedAddress/SavedAddressUseCase.java
package com.lattis.ellipse.domain.interactor.SavedAddress;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.SavedAddress;
import com.lattis.ellipse.domain.repository.SavedAddressRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

public class SavedAddressUseCase extends UseCase<Boolean>

{

    private SavedAddressRepository savedAddressRepository;
    private List<SavedAddress>  savedAddresses;

    @Inject
    public SavedAddressUseCase(ThreadExecutor threadExecutor,
                           PostExecutionThread postExecutionThread,
                           SavedAddressRepository savedAddressRepository) {
        super(threadExecutor, postExecutionThread);
        this.savedAddressRepository = savedAddressRepository;
    }

    public SavedAddressUseCase withSavedAddress(List<SavedAddress> savedAddresses) {
        this.savedAddresses = savedAddresses;
        return this;
    }


    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return this.savedAddressRepository.saveAddresses(savedAddresses);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/saveaddress/SavedAddressUseCase.kt
package com.lattis.domain.usecase.saveaddress

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.SavedAddressRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.SavedAddress
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class SavedAddressUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val savedAddressRepository: SavedAddressRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var savedAddresses: List<SavedAddress>? = null
    fun withSavedAddress(savedAddresses: List<SavedAddress>?): SavedAddressUseCase {
        this.savedAddresses = savedAddresses
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return savedAddressRepository.saveAddresses(savedAddresses!!)
    }

}